pkg/content/ported.go

```diff
@@ -160,7 +162,11 @@ func fetchHandler(ingester content.Ingester, fetcher remotes.Fetcher) images.Han
 			return nil, fmt.Errorf("%v not supported", desc.MediaType)
 		default:
 			_, err, _ := fetchSingleflight.Do(string(desc.Digest), func() (interface{}, error) {
-				return nil, remotes.Fetch(ctx, ingester, fetcher, desc)
+				err := remotes.Fetch(ctx, ingester, fetcher, desc)
```

Comment:
The `err` should be handled first.

Reply:
When ```remotes. Fetch``` handled first, if the layer exits locally, the ```err``` will not be ```nil```, so I return ```err``` later to make sure ```UpdateTime``` will be called.

Reply:
Ok, but if the layer does not exist in local and a network error occurs in `remotes.Fetch`, users can't track the real error.

Reply:
Maybe we can check the err type？

Reply:
Fixed.